plugins/VstEffect/VstEffect.cpp

```diff
@@ -76,46 +78,30 @@ VstEffect::VstEffect( Model * _parent,
 
 
 
-bool VstEffect::processAudioBuffer( SampleFrame* _buf, const fpp_t _frames )
+double VstEffect::processImpl(SampleFrame* buf, const fpp_t frames)
 {
-	if( !isEnabled() || !isRunning () )
+	assert(m_plugin != nullptr);
+	assert(frames <= DEFAULT_BUFFER_SIZE);
+	static thread_local auto tempBuf = std::array<SampleFrame, DEFAULT_BUFFER_SIZE>();
```

Comment:
This should work for buffer sizes less than 256 too. `DEFAULT_BUFFER_SIZE` is currently the effective maximum buffer size (since `frames` should never be greater than it). I just wanted to express that assumption with an assert. To my knowledge we do not have any better-named `MAX_BUFFER_SIZE` constant to use instead.

Reply:
> This should work for buffer sizes less than 256 too.

And what if it is greater than 256? (This is a possibility in the near future, which is my point)

> DEFAULT_BUFFER_SIZE is currently the effective maximum buffer size (since frames should never be greater than it)

And what I am trying to tell you is that this is suspect to change. The way LMMS handles its buffers by splitting the buffer into chunks, each 256 sample frames each, has created a number of issues and is generally not how a DAW is supposed to deal with varying buffer sizes.

The only reason why this chunking process is here is because of granularity. A good number of our parameters only update once per buffer, so we needed to process the audio in chunks where each chunk wasn't too big or too small, and it was decided that we should process the buffers into chunks of 256. However, as I mentioned this has created a number of problems and the DAW should be expected to use the buffer size as is while also updating any parameters or doing any operations necessary at their expected rates (i.e., not necessarily once per buffer).

That being said, the maximum buffer size is not truly 256. The maximum buffer size is actually 4096 currently, but LMMS does this chunking process which I am trying to remove.

The minute LMMS no longer always uses a fixed buffer size of 256, this will immediately break the VST processing, so I am suggesting that you use `ArrayVector` here with the maximum size being `4096`. We can only safely guarantee at all times, regardless of any changes that happen, that `frames` will always be less than or equal to the maximum buffer size allowed, which is `4096` as of right now.

Reply:
Ok, I think I'll add this constant to `AudioEngine.h`:
```c++
constexpr fpp_t MAXIMUM_BUFFER_SIZE = 4096;
```

I'd rather keep using `std::array` because I just need a simple scratch pad that is guaranteed to have enough space.